### ISSUE TITLE
Issue with up-or-search

### ISSUE THREAD

**Issue by geier** [fish-shell/fish-shell#3433]
- [x] Have you checked if problem occurs with [fish 2.3.1](/fish-shell/fish-shell/releases/tag/2.3.1)?
- [x] Tried fish without third-party customizations *(check `sh -c 'env HOME=$(mktemp -d) fish'`)*?

**fish version installed** *(`fish --version`)*: 2.3.1 (and also tried 91a311c and 2.0.0).

**OS/terminal used**: Gnome Terminal and xterm on Ubuntu 16.04.1 LTS as well as FreeBSD 10.3

When having typed the beginning of a command, fish auto-suggests (fainted, right side of the cursor) the last command that begins with the already typed part. I can complete it with <kbd>tab</kbd>, <kbd>ctrl</kbd> + <kbd>f</kbd> or <kbd>right</kbd>.
But when pressing <kbd>up</kbd> it goes to the matching command *before* that and then backwards in history as expected, but the very last matching command doesn't get shown.
I would expect that pressing <kbd>up</kbd> only once should display the last matching command. Is this behavior intentionally?

I've made an [asciinema recording of the issue](https://asciinema.org/a/3bdzg7cu8dje5lqktnscie7vl) that might help illustrate the issue.

**Comment by karolyi** [comment]
it has been always like that, ever since I use fish. you'll get the first match by pressing <kbd>right</kbd> after started typing

**Comment by faho** [comment]
This is a duplicate of #405.

**Comment by geier** [comment]
@faho thanks for pointing that out (I really tried to search for it...)
